src/Uno.UI/UI/Xaml/Controls/MediaPlayerElement/MediaTransportControls.cs

```diff
@@ -575,56 +747,374 @@ private void OnRootGridTapped(object sender, TappedRoutedEventArgs e)
+					break;
+				case var _ when property == IsStopButtonVisibleProperty:
+					BindVisibility(m_tpStopButton, IsStopButtonVisible);
+					break;
+				case var _ when property == IsStopEnabledProperty:
+					BindIsEnabled(m_tpStopButton, IsStopEnabled);
+					break;
+				case var _ when property == IsSeekBarVisibleProperty:
+					BindVisibility(_timelineContainer, IsSeekBarVisible);
+					break;
+				case var _ when property == IsSeekEnabledProperty:
+					BindIsEnabled(_timelineContainer, IsSeekEnabled);
+					break;
+				case var _ when property == IsSkipForwardButtonVisibleProperty:
+					BindVisibility(m_tpSkipForwardButton, IsSkipForwardButtonVisible);
+					break;
+				case var _ when property == IsSkipForwardEnabledProperty:
+					BindIsEnabled(m_tpSkipForwardButton, IsSkipForwardEnabled);
+					break;
+				case var _ when property == IsSkipBackwardButtonVisibleProperty:
+					BindVisibility(m_tpSkipBackwardButton, IsSkipBackwardButtonVisible);
+					break;
+				case var _ when property == IsSkipBackwardEnabledProperty:
+					BindIsEnabled(m_tpSkipBackwardButton, IsSkipBackwardEnabled);
+					break;
+				case var _ when property == IsNextTrackButtonVisibleProperty:
+					BindVisibility(m_tpNextTrackButton, IsNextTrackButtonVisible);
+					break;
+				case var _ when property == IsPreviousTrackButtonVisibleProperty:
+					BindVisibility(m_tpPreviousTrackButton, IsPreviousTrackButtonVisible);
+					break;
+				case var _ when property == IsCompactOverlayButtonVisibleProperty:
+					BindVisibility(m_tpCompactOverlayButton, IsCompactOverlayButtonVisible);
+					break;
+				case var _ when property == IsCompactOverlayEnabledProperty:
+					BindIsEnabled(m_tpCompactOverlayButton, IsCompactOverlayEnabled);
+					break;
+			}
 
-		private static void OnShowAndHideAutomaticallyChanged(DependencyObject dependencyObject, DependencyPropertyChangedEventArgs args)
+			void BindVisibility(FrameworkElement target, bool value)
+			{
+				if (target is { })
+				{
+					target.Visibility = value ? Visibility.Visible : Visibility.Collapsed;
+				}
+			}
+			void BindIsEnabled(FrameworkElement target, bool value)
+			{
+				if (target is { })
+				{
+					target.IsEnabled = value;
+				}
+			}
+		}
+		private void OnShowAndHideAutomaticallyChanged()
 		{
-			if ((bool)args.NewValue)
+			if (ShowAndHideAutomatically)
 			{
-				((MediaTransportControls)dependencyObject).ResetControlsVisibilityTimer();
+				ResetControlsVisibilityTimer();
 			}
 			else
 			{
-				((MediaTransportControls)dependencyObject).CancelControlsVisibilityTimer();
+				CancelControlsVisibilityTimer();
+			}
+		}
+
+		// visual states
+		private void UpdateAllVisualStates(bool useTransition = true)
+		{
+			UpdateControlPanelVisibilityStates(useTransition);
+			UpdateMediaStates(useTransition);
+			//UpdateAudioSelectionAvailablityStates(useTransition);
+			//UpdateCCSelectionAvailablityStates(useTransition);
+			//UpdateFocusStates(useTransition);
```

Comment:
Normal that these are kept commented ?
If so would be nice to have a comment with a bit more details.

Reply:
thats all visual states we have, their relevant parts are not implemented yet so there is nothing to do there atm

Reply:
Just keep a comment similar to "Relevant parts are not implemented yet" then will be helpful I think